### PR TITLE
Add Arabic translation for README

### LINKED ISSUE #29045: [i18n-ar] Translating docs to Arabic
Hi!

Let's bring the documentation to all the Arabic-speaking community 🌐 

Who would want to translate? Please follow the 🤗 [TRANSLATING guide](https://github.com/huggingface/transformers/blob/main/docs/TRANSLATING.md). Here is a list of the files ready for translation. Let us know in this issue if you'd like to translate any, and we'll add your name to the list.

Some notes:

* Please translate using an informal tone (imagine you are talking with a friend about transformers 🤗).
* Please translate in a gender-neutral way.
* Add your translations to the folder called `<languageCode>` inside the [source folder](https://github.com/huggingface/transformers/tree/main/docs/source).
* Register your translation in `<languageCode>/_toctree.yml`; please follow the order of the [English version](https://github.com/huggingface/transformers/blob/main/docs/source/en/_toctree.yml).
* Once you're finished, open a pull request and tag this issue by including #issue-number in the description, where issue-number is the number of this issue. Please ping @stevhliu and @MKhalusova for review.
* 🙋 If you'd like others to help you with the translation, you can also post in the 🤗 [forums](https://discuss.huggingface.co/).

## Get Started section
- [ ] [readme.md](https://github.com/huggingface/transformers/edit/main/README.md)
- [ ] [index.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/index.md) 
- [ ] [quicktour.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/quicktour.md)
- [ ] [installation.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/installation.md)

## Tutorial section
- [ ] [pipeline_tutorial.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/pipeline_tutorial.md)
- [ ]  [autoclass_tutorial.md](https://github.com/huggingface/transformers/blob/master/docs/source/autoclass_tutorial.md)
- [ ]  [preprocessing.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/preprocessing.md)
- [ ]  [training.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/training.md)
- [ ]  [accelerate.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/accelerate.md)
- [ ]  [model_sharing.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/model_sharing.md)
- [ ]  [multilingual.md](https://github.com/huggingface/transformers/blob/main/docs/source/en/multilingual.md)

### PR DESCRIPTION
**Pull request by strikoder**
# What does this PR do?
This PR introduces the Arabic translation of the README file.

## Before submitting
- [x] This PR improves the docs.

## Fixes #29045 

 @stevhliu 
@MKhalusova

### CODE REVIEW

File: README_ar.md [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+```
+
+هنا، نحصل على قائمة الأجسام التي تم اكتشافها في الصورة، مع صندوق يحيط بالجسم ودرجة ثقة. هذه هي الصورة الأصلية على اليسار، مع عرض التوقعات على اليمين:
+
+<h3 align="center">
+    <a><img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/coco_sample.png" width="400"></a>
+    <a><img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/coco_sample_post_processed.png" width="400"></a>
+</h3>
+
+يمكنك التعرف على المزيد حول المهام المدعومة من قبل واجهة برمجة التطبيقات `pipeline` في [هذا الدليل](https://huggingface.co/docs/transformers/task_summary).
```
Comment (stevhliu) [inline README_ar.md:197]: Link format wrong here too

File: README_ar.md [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+>>> tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+>>> model = TFAutoModel.from_pretrained("bert-base-uncased")
+
+>>> inputs = tokenizer("Hello world!", return_tensors="tf")
+>>> outputs = model(**inputs)
+```
+
+المُحوّل (Tokenizer) مسؤول عن كل عمليات المعالجة المسبقة التي يتوقعها النموذج المُدرب مسبقًا ويمكن استدعاؤه مباشرةً على سلسلة نصية واحدة (كما في الأمثلة أعلاه) أو قائمة. سيُخرج قاموسًا يمكنك استخدامه في الكود اللاحق أو ببساطة تمريره مباشرةً إلى نموذجك باستخدام عامل فك الحزم **.
+
+النموذج نفسه هو [وحدة `nn.Module` من Pytorch](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) أو [`tf.keras.Model` من TensorFlow](https://www.tensorflow.org/api_docs/python/tf/keras/Model) (حسب النظام الأساسي الذي تستخدمه) والذي يمكنك استخدامه كالمعتاد. [يشرح هذا الدليل](https://huggingface.co/docs/transformers/training) كيفية دمج مثل هذا النموذج في حلقة تدريب كلاسيكية لـ PyTorch أو TensorFlow، أو كيفية استخدام واجهة برمجة التطبيقات `Trainer` لدينا للضبط الدقيق بسرعة على مجموعة بيانات جديدة.
```
Comment (stevhliu) [inline README_ar.md:225]: The link format to the training tutorial is incorrect

File: README_ar.md [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+4. تخصيص نموذج أو مثال لاحتياجاتك بسهولة:
+    - نوفر أمثلة لكل هندسة لإعادة إنتاج النتائج التي نشرها مؤلفوها الأصليون.
+    - تتم تعرض دواخل النموذج بشكل متسق قدر الإمكان.
+    - يمكن استخدام ملفات النموذج بشكل مستقل عن المكتبة للتجارب السريعة.
+    
+## لماذا لايجب علي استخدام Transformers؟
+
+- هذه المكتبة ليست صندوق أدوات معياري من الكتل البنائية للشبكات العصبية. الكود الموجود في ملفات النموذج ليس معاد تنظيمه مع تجريدات إضافية عن عمد، حتى يتمكن الباحثون من التكرار بسرعة على كل من النماذج دون الغوص في تجريدات/ملفات إضافية.
+- واجهة برمجة التطبيقات للتدريب ليست مخصصة للعمل مع أي نموذج ولكنها مُحسنة للعمل مع النماذج التي توفرها المكتبة. لأجل حلقات التعلم الآلي العامة، يجب استخدام مكتبة أخرى (ربما، [Accelerate](https://huggingface.co/docs/accelerate)).
+- بينما نسعى لتقديم أكبر عدد ممكن من حالات الاستخدام، السكربتات في [مجلد الأمثلة](https://github.com/huggingface/transformers/tree/main/examples) لدينا هي مجرد ذلك: أمثلة. من المتوقع أنها لن تعمل مباشرةً على مشكلتك الخاصة وأنه سيتطلب منك تغيير بضعة أسطر من الكود لتكييفها مع احتياجاتك.
```
Comment (stevhliu) [inline README_ar.md:254]: Link here also won't work

File: README_ar.md [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+
+## التثبيت
+
+### بواسطة pip
+
+تم اختبار هذا الريبو على +Python 3.8+، Flax 0.4.1+، PyTorch 1.11، و +TensorFlow 2.6.
+
+يجب تثبيت 🤗 Transformers في [بيئة افتراضية](https://docs.python.org/3/library/venv.html). إذا كنت غير مألوف ببيئات Python الافتراضية، راجع [دليل المستخدم](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/).
```
Comment (stevhliu) [inline README_ar.md:262]: Link to the Python virtual environment guide won't work

File: README_ar.md [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+ثم، ستحتاج إلى تثبيت واحد على الأقل من Flax، PyTorch، أو TensorFlow.
+يرجى الرجوع إلى [صفحة تثبيت TensorFlow](https://www.tensorflow.org/install/)، [صفحة تثبيت PyTorch](https://pytorch.org/get-started/locally/#start-locally) و/أو [Flax](https://github.com/google/flax#quick-install) و [Jax](https://github.com/google/jax#installation) صفحات التثبيت لمعرفة الأمر المحدد لتثبيتها على منصتك.
+
+عند تثبيت أحد هذه الخلفيات، يمكن تثبيت 🤗 Transformers باستخدام pip على النحو التالي:
+
+```bash
+pip install transformers
+```
+
+إذا أردت التلاعب بالأمثلة أو كنت بحاجة إلى أحدث إصدار من الكود ولا يمكنك الانتظار لإصدار جديد، يجب عليك [تثبيت المكتبة من المصدر](https://huggingface.co/docs/transformers/installation#installing-from-source).
```
Comment (stevhliu) [inline README_ar.md:275]: This link also won't work

File: README_ar.md [hunk 6]
```diff
@@ -0,0 +1,602 @@
...
+
+```shell script
+conda install conda-forge::transformers
+```
+
+> **_ملاحظة:_** تثبيت `transformers` من قناة `huggingface` أصبح مهجورًا.
+
+اتبع صفحات التثبيت لـ Flax، PyTorch أو TensorFlow لمعرفة كيفية تثبيتها باستخدام conda.
+
+> **_ملاحظة:_** على نظام التشغيل Windows، قد يُطلب منك تفعيل وضع المطور للاستفادة من التخزين المؤقت. إذا لم يكن هذا خيارًا لك، يرجى إعلامنا في [هذه المشكلة](https://github.com/huggingface/huggingface_hub/issues/1062).
```
Comment (stevhliu) [inline README_ar.md:289]: Same with link to the issue

**Review by stevhliu — COMMENTED** [review]
Thanks for starting an Arabic translation! My main feedback is to double-check the Markdown links are correctly formatted (`[]()`) so the links work. Some of the URLs are currently separated from the text in the brackets so I don't think the links will work.

It would also help if you could find someone who can read Arabic to review your work because I can't read Arabic! 😅

File: README_ar.md [hunk 7]
```diff
@@ -0,0 +1,602 @@
...
+يمكن تطبيق هذه النماذج على:
+
+* 📝 النص، لمهام مثل تصنيف النصوص، استخراج المعلومات، الإجابة عن الأسئلة، الخلاصة، الترجمة، وتوليد النص، بأكثر من 100 لغة.
+* 🖼️ الصور، لمهام مثل تصنيف الصور، التعرّف على الأجسام، والتجزئة.
+* 🗣️ الصوت، لمهام مثل التعرف على الكلام وتصنيف الصوت.
+
+يمكن لنماذج Transformer أيضًا أداء مهام على **عدة وسائط مجتمعة**، مثل الإجابة عن الأسئلة المتعلقة بالجداول، التعرف الضوئي على الحروف، استخراج المعلومات من الوثائق الممسوحة ضوئيًا، تصنيف الفيديو، والإجابة عن الأسئلة البصرية.
+
+
+🤗 Transformers يوفر واجهات برمجة تطبيقات لتنزيل واستخدام هذه النماذج المدربة مسبقًا بسرعة على نص معين، وصقلها على مجموعات البيانات الخاصة بك ثم مشاركتها مع المجتمع على [مركز النماذج](https://huggingface.co/models) الخاص بنا. في الوقت نفسه، كل وحدة بايثون تحدد هندسة هي مستقلة تمامًا ويمكن تعديلها لتمكين تجارب البحث السريع.
```
[the remark this answers is not in the document]
Comment (Strikoder) [reply README_ar.md:80]: Thank you for your feedback. The link does work as intended when viewed in the Markdown file and not checking the actual Markdown code. The formatting issue, including the unexpected new line, arises from using Arabic text, an RTL language, which can affect Markdown syntax. Rest assured, the link's functionality is unaffected. I appreciate your understanding.

File: README_ar.md [hunk 8]
```diff
@@ -0,0 +1,602 @@
...
+
+🤗 Transformers مدعومة من أشهر ثلاث مكتبات التعلم العميق — [Jax](https://jax.readthedocs.io/en/latest/)، [PyTorch](https://pytorch.org/) و [TensorFlow](https://www.tensorflow.org/) — مع تكامل سلس بينها. من السهل تدريب نماذجك مع واحدة قبل تحميلها للتنبؤ بالآخر.
+
+## العروض التوضيحية عبر الإنترنت
+
+يمكنك اختبار معظم نماذجنا مباشرةً على صفحاتها من [مركز النماذج](https://huggingface.co/models). نقدم أيضًا [استضافة نماذج خاصة، الإصدارات، وواجهة برمجة التطبيقات للتنبؤ](https://huggingface.co/pricing) للنماذج العامة والخاصة.
+
+يمكنك اختبار معظم نماذجنا مباشرةً على صفحاتها من [مركز النماذج](https://huggingface.co/models). نقدم أيضًا [استضافة نماذج خاصة، الإصدارات، وواجهة برمجة التطبيقات للتنبؤ](https://huggingface.co/pricing) للنماذج العامة والخاصة.
```
[the remark this answers is not in the document]
Comment (Strikoder) [reply README_ar.md]: Yup, it's a duplicate paragraph, I will remove it, thank you!

**Comment by Strikoder** [comment]
Here are some screenshots from the actual files, I double checked all the issues you raised, everything seems fine
screenshot_2024-02-20-211431

screenshot_2024-02-20-211429

screenshot_2024-02-20-211422

screenshot_2024-02-20-211413
screenshot_2024-02-20-211408

File: README_ar.md [hunk 9]
```diff
@@ -0,0 +1,602 @@
...
+يمكن لنماذج Transformer أيضًا أداء مهام على **عدة وسائط مجتمعة**، مثل الإجابة عن الأسئلة المتعلقة بالجداول، التعرف الضوئي على الحروف، استخراج المعلومات من الوثائق الممسوحة ضوئيًا، تصنيف الفيديو، والإجابة عن الأسئلة البصرية.
+
+
+🤗 Transformers يوفر واجهات برمجة تطبيقات لتنزيل واستخدام هذه النماذج المدربة مسبقًا بسرعة على نص معين، وصقلها على مجموعات البيانات الخاصة بك ثم مشاركتها مع المجتمع على [مركز النماذج](https://huggingface.co/models) الخاص بنا. في الوقت نفسه، كل وحدة بايثون تحدد هندسة هي مستقلة تمامًا ويمكن تعديلها لتمكين تجارب البحث السريع.
+
+🤗 Transformers مدعومة من أشهر ثلاث مكتبات التعلم العميق — [Jax](https://jax.readthedocs.io/en/latest/)، [PyTorch](https://pytorch.org/) و [TensorFlow](https://www.tensorflow.org/) — مع تكامل سلس بينها. من السهل تدريب نماذجك مع واحدة قبل تحميلها للتنبؤ بالآخر.
+
+## العروض التوضيحية عبر الإنترنت
+
+يمكنك اختبار معظم نماذجنا مباشرةً على صفحاتها من [مركز النماذج](https://huggingface.co/models). نقدم أيضًا [استضافة نماذج خاصة، الإصدارات، وواجهة برمجة التطبيقات للتنبؤ](https://huggingface.co/pricing) للنماذج العامة والخاصة.
```
[the remark this answers is not in the document]
Comment (Strikoder) [reply README_ar.md:86]: screenshot_2024-02-20-211857

File: README_ar.md (hunk 1 above)
Comment (Strikoder) [reply README_ar.md:195]: screenshot_2024-02-20-211857

Comment (Strikoder) [reply README_ar.md:195]: screenshot_2024-02-20-211957

File: README_ar.md (hunk 3 above)
Comment (Strikoder) [reply README_ar.md:252]: screenshot_2024-02-20-212120

File: README_ar.md (hunk 2 above)
Comment (Strikoder) [reply README_ar.md:223]: screenshot_2024-02-20-212157

File: README_ar.md (hunk 4 above)
Comment (Strikoder) [reply README_ar.md:260]: screenshot_2024-02-20-212226

File: README_ar.md (hunk 6 above)
Comment (Strikoder) [reply README_ar.md:287]: image

File: README_ar.md (hunk 5 above)
Comment (Strikoder) [reply README_ar.md:273]: image

**Comment by Strikoder** [comment]
I removed the duplicates and double checked the Markdown issue, all links work properly.
I don't really know who speaks Arabic, could you please help me with that?

**Comment by Strikoder** [comment]
I removed the duplicates and double checked the Markdown issue, all links work properly.
I don't really know who speaks Arabic at Hugging Face, could you please help me with that?

**Comment by Strikoder** [comment]
I've addressed the duplicates and verified the Markdown formatting; all links are now functioning correctly. Screenshots documenting each issue you highlighted have been attached for reference. 

Regarding the Arabic langauge, I don't really know who speaks Arabic at Hugging Face, could you please help me with that?

**Comment by stevhliu** [comment]
Great, thanks!

I would join our [Discord](https://discord.gg/hugging-face-879548962464493619) and see if any community members there would be interested in reviewing the translation. You can also try asking on our [forums](https://discuss.huggingface.co/) as well :)

**Comment by AbdalrahmanAliElnashar** [comment]
I am Egyptian and I speak Arabic.
Everything is correct 💯 
the work of @Strikoder is right and important for Arabic Language.
thanks to you 👍🏻

**Pull request closed without merging**